kedro-org/kedro-viz
# Remove Python Upper Bound Requirements

## Description

Resolves #1505 

## Development notes

https://github.com/kedro-org/kedro/pull/2742

## QA notes

1. Navigate to kedro-viz project folder 
2. Checkout branch - `chore/remove-upper-bound`

**Local conda env setup for python 3.11 -**
  - Create a conda environment with python 3.11 - 
  
    ```shell
    conda create -n yourenvname python=3.11
    ```  
  - Activate the created conda environment - 
  
     ```shell
    conda activate yourenvname
    ```  

**Testing kedro-viz dev server -**

1. Install package dependencies for kedro-viz -

      ```python
    pip3 install -r package/test_requirements.txt -r demo-project/src/docker_requirements.txt
    ``` 

2. Run backend server by executing the command - 
     
     ```shell
    make run
    ```
3. You should observe a warning as below -

    ```shell
    Org/kedro-viz/package/kedro_viz/__init__.py:16: KedroVizPythonVersionWarning: Please be                     
                             advised that Kedro Viz is not yet fully                                                                     
                                     compatible with the Python version you are currently using.                                         
                               warnings.warn(      
    ```

**Testing kedro-viz as a package -**

1. Build frontend

   ```shell
    make build    
    ```
    
2. Install local Kedro-viz package - 

    ```python
    pip install -e package
    ```

3. Navigate to demo-project -  
  
    ```shell
    cd demo-project
    ```
4. Execute kedro viz command to open up in the browser [here](http://127.0.0.1:4141/)
 
   ```shell
    kedro viz
    ```
5.  You should observe a warning as below -

    ```shell
    Org/kedro-viz/package/kedro_viz/__init__.py:16: KedroVizPythonVersionWarning: Please be                     
                             advised that Kedro Viz is not yet fully                                                                     
                                     compatible with the Python version you are currently using.                                         
                               warnings.warn(      

## Checklist



<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1506"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
